helix-core/src/main/java/org/apache/helix/task/TaskDriver.java

```diff
@@ -804,7 +803,9 @@ private void validateConfigsForTaskModifications(String workflowName, String job
     }
 
     if ((taskConfig.getCommand() == null) == (jobConfig.getCommand() == null)) {
-      throw new HelixException("Command must exist in either job or task, not both!");
+      throw new HelixException(String
+          .format("Command must exist in either job %s or task %s, not both!", jobName,
+              taskConfig.getId()));
```

Comment:
This may confusing. Originally it tells us the command should be either in job config or task config. But when you dd job name and task id can distract the focus. 

You can change it as: "Command must exist in either jobconfig (jobname) or taskconfig (taskid)" something.

Reply:
TFTR. Updated.